src/main/java/org/commcare/xml/SessionDatumParser.java

```diff
@@ -132,10 +133,11 @@ private RemoteQueryDatum parseRemoteQueryDatum()
             } else if ("prompt".equals(tagName)) {
                 String key = parser.getAttributeValue(null, "key");
                 userQueryPrompts.put(key, new QueryPromptParser(parser).parse());
+            } else if ("locale".equals(tagName)) {
```

Comment:
Was it not feasible to do this before the loop, as we chatted about?

I was thinking something like [this](https://github.com/dimagi/commcare-core/blob/ce29258248030b230743b55f9501e5abd22f37f9/src/main/java/org/commcare/xml/DetailParser.java#L43-L51), only without the `checkNode` since the title isn't guaranteed to be there (until apps are migrated).